### PR TITLE
feat(`x/marketmap`): params type

### PR DESCRIPTION
**Pull request by aljo242**
Closes BLO-1030

### CODE REVIEW

File: x/mm2/types/params.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	}, nil
+}
+
+// ValidateBasic performs stateless validation of the Params.
+func (p *Params) ValidateBasic() error {
+	if p.MarketAuthorities == nil {
+		return fmt.Errorf("cannot create Params with nil authority")
+	}
+
+	seenAuthorities := make(map[string]struct{}, len(p.MarketAuthorities))
```
Comment (g3ndo9kar9) [inline x/mm2/types/params.go:38]: we perform this check so frequently (uniqueness of items in an array), perhaps it's useful for us to make a generic helper in `/lib` and start using that

File: proto/slinky/mm2/v1/params.proto [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+// Params defines the parameters for the x/marketmap module.
+message Params {
+  // MarketAuthorities is the list of authority accounts that are able to
+  // control updating the marketmap.
+  repeated string market_authorities = 1;
+
+  // Version is the schema version for the MarketMap data structure and query
+  // response.
+  uint64 version = 2;
```
Comment (g3ndo9kar9) [inline proto/slinky/mm2/v1/params.proto]: We're putting the version in params? I thought we'd be putting it in the market-map? How else would we be correlating between the mm data-structure + the version?

File: proto/slinky/mm2/v1/params.proto [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+}
```
Comment (g3ndo9kar9) [inline proto/slinky/mm2/v1/params.proto]: nit: newline at EOF

File: proto/slinky/mm2/v1/params.proto (hunk 2 above)
Comment (davidterpay) [reply proto/slinky/mm2/v1/params.proto]: Can we have this be a string which is essentially just the tag?

File: x/mm2/types/params.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+)
+
+const (
+	// DefaultVersion is the default value for the Version Param.
+	DefaultVersion = 0
```
Comment (davidterpay) [inline x/mm2/types/params.go]: following up on the other point, we should just hard code this to be the tag when the mm is ready.

Comment (davidterpay) [reply x/mm2/types/params.go]: presumably `v1.0.0`

Comment (aljo242) [reply x/mm2/types/params.go]: What do you mean by this?

File: proto/slinky/mm2/v1/params.proto (hunk 2 above)
Comment (aljo242) [reply proto/slinky/mm2/v1/params.proto]: @davidterpay How would that be set? My thinking is that this was the schema version of the market map struct. Perhaps this isn't even a useful thing to have. We can do versioning things in the sidecar based on the tag

Comment (davidterpay) [reply proto/slinky/mm2/v1/params.proto]: The marketmap scheme is likely not going to change very often but we should have a way of signaling validators when to update their side-car with a new tag/version. imo you kinda accomplish both in the same case, any schema change to the market map is going require a new side car version anyways

Comment (davidterpay) [reply proto/slinky/mm2/v1/params.proto]: im assuming once this is ready we just set the default to v1.0.0

File: x/mm2/types/params.go (hunk 4 above)
Comment (davidterpay) [reply x/mm2/types/params.go]: the release we cut with the mm is probably going to be v1.0.0

**Comment by aljo242** [comment]
@nivasan1 @davidterpay removing `Version` field for now until we get alignment on how / if we will use this

**Review by davidterpay — APPROVED** [review]

File: x/mm2/types/params.go (hunk 1 above)
Comment (aljo242) [reply x/mm2/types/params.go:38]: https://linear.app/skip/issue/BLO-1059/create-lib-function-for-checking-uniqueness-of-items-in-a-list

File: proto/slinky/mm2/v1/params.proto [hunk 5]
```diff
@@ -0,0 +1,11 @@
+syntax = "proto3";
+package slinky.mm2.v1;
+
+option go_package = "github.com/skip-mev/slinky/x/mm2/types";
+
+// Params defines the parameters for the x/marketmap module.
+message Params {
+  // MarketAuthorities is the list of authority accounts that are able to
+  // control updating the marketmap.
+  repeated string market_authorities = 1;
```
Comment (Eric-Warehime) [inline proto/slinky/mm2/v1/params.proto:10]: Maybe we can do this in a followup:

I thought our idea was to have a 2 tiered permissions system in which participants could update the mm, and a single "admin" authority could add/remove participants.

File: x/mm2/types/params.go [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+func DefaultParams() Params {
+	return Params{
+		MarketAuthorities: []string{DefaultMarketAuthority},
+	}
+}
+
+// NewParams returns a new Params instance.
+func NewParams(authorities []string) (Params, error) {
+	if authorities != nil {
+		return Params{}, fmt.Errorf("cannot create Params with nil authority")
```
Comment (Eric-Warehime) [inline x/mm2/types/params.go]: Not sure how this works, shouldn't this be `if authorities == nil` ?

File: proto/slinky/mm2/v1/params.proto (hunk 5 above)
Comment (aljo242) [reply proto/slinky/mm2/v1/params.proto:10]: yeah - I added this in a follow up https://github.com/skip-mev/slinky/pull/293

**Review by Eric-Warehime — APPROVED** [review]

**Pull request merged**